tiledb/api/src/query/read/aggregate/mod.rs

```diff
@@ -0,0 +1,1054 @@
+            assert!(matches!(
+                e,
+                Err(TileDBError::Datatype(DatatypeErrorKind::TypeMismatch {
+                    tiledb_type: Datatype::Float64,
+                    ..
+                }))
+            ));
+        }
+
+        Ok(())
+    }
+
+    /// When this test fails, update `impl Arbitrary for AggregateFunction`
+    #[test]
+    fn sc_53791_null_count_on_var_attribute() -> TileDBResult<()> {
+        let c = Context::new().unwrap();
+
+        let schema = {
+            let domain = {
+                let dimension = DimensionBuilder::new(
+                    &c,
+                    "d",
+                    Datatype::UInt64,
+                    [0u64, 100u64],
+                )?
+                .build();
+                DomainBuilder::new(&c)?.add_dimension(dimension)?.build()
+            };
+            let attribute = AttributeBuilder::new(&c, "a", Datatype::UInt64)?
+                .nullability(true)?
+                .cell_val_num(CellValNum::Var)?
+                .build();
+
+            SchemaBuilder::new(&c, ArrayType::Sparse, domain)?
+                .add_attribute(attribute)?
+                .build()?
+        };
+
+        let test_uri = tiledb_test_utils::get_uri_generator()
+            .map_err(|e| Error::Other(e.to_string()))?;
+
+        let uri = test_uri
+            .with_path("sc_53791")
+            .map_err(|e| Error::Other(e.to_string()))?;
+
+        Array::create(&c, &uri, schema)?;
+
+        // insert a cell
+        {
+            let values_d = vec![0u64];
+            let values_a = vec![vec![0u64]];
+
+            let a = Array::open(&c, &uri, Mode::Write)?;
+            let q = WriteBuilder::new(a)?
+                .data("d", &values_d)?
+                .data("a", &values_a)?
+                .build();
+
+            q.submit()?;
+            q.finalize()?;
+        }
+
+        // try query
+        let a = Array::open(&c, &uri, Mode::Read)?;
+
+        let mut q = ReadBuilder::new(a)?
+            .layout(QueryLayout::Unordered)?
+            .null_count("a")?
+            .build();
+        let r = q.execute();
+        assert!(matches!(r, Err(Error::LibTileDB(_))));
+
+        Ok(())
+    }
+
+    /// Test running min/max aggregates on empty input.
+    #[test]
+    fn min_max_non_nullable_empty_input() -> TileDBResult<()> {
+        let a = TestArray::new(
+            "min_max_empty_input",
+            Rc::new(
+                crate::tests::examples::quickstart::Builder::new(
+                    ArrayType::Sparse,
+                )
+                .build(),
+            ),
+        )?;
+
+        let mut q = ReadBuilder::new(a.for_read()?)?
+            .min::<i32>("a")?
+            .max::<i32>("a")?
+            .build();
+        let (a_max, (a_min, _)) = q.execute()?;
+
+        // This is deliberately a wrong result.
+        // We capture it here to track SC-54468.
```

Comment:
Super minor nit, I really like these "add tests that will fail when bug is fixed" approach. I think we should consider standardizing the `sc_#####_` prefix for tests that exist for this reason.

Reply:
Will do

Reply:
4c1e4c0165